### DISCUSSION TITLE
CoApp Win7: "Error writing temporary file"

### DISCUSSION THREAD

**Discussion by andherson17** [aclap-dev/video-downloadhelper#1053] [Help and Q&A]
### Your question

The Win10 version of CoApp 'vdhcoapp-windows-x86_64-installer.exe' installed on my Win10 notebook and works fine. But trying to install 'vdhcoapp-win7-x86_64-installer.exe' and all older versions to my preferred old Win7 Ultimate PC brings up the error: 

"Error writing temporary file. Make sure your temp folder is valid." 

The AppData\Local\Temp folder is there. I've emptied it successfully with Disk Cleanup. and had no problems experimenting with adding and deleting a few files manually. It seems I solved this with other installers in the past by running the installer from the command line. But that still gets the same error with this one. Has anyone seen this and found any a solution?
Kind thanks.

### Addon & CoApp version and details

File description: Video DownloadHelper companion app
Type: Application
File version: 2.0.19.0
Product name: DownloadHelper CoApp
Copyright: ACLAP
Size: 37.0 MB
Date modified: 7/24/2024 6:66 PM
Language: English (United States)

**Comment by andherson17** [comment]
I turned off my antivirus, copied the executable to a root folder, shortened the filename, ran it from the command line and it installed. But the Video DownloadHelper browser plugin complained saying CoApp needed updating. Version 2.0.19.0 seems to be the latest 64 bit version.

So I downloaded the 32 bit version 'vdhcoapp-win7-i686-installer.exe'. Even with antivirus running, it threw up that tamp folder error, but it installed from the command line with a shortened filename as before, The Chrome browser addon recognized it and downloaded the Prime Video from my account there. But as was the case on my Win10 notebook, the final competed video download was corrupt. 

Medialnfo reports the video and audio components properly, but playback is just snowy garbage from end to end.

**Comment by Mattmjs** [comment]
I think windows 7 is too old now, already 4 years out of support. It might come a time that VDH will eventually drop support
for this OS. This is my opinion though since I'm just another user of the software like you. Is there a reason you still need windows 7 ?
In your situation I'd switch to windows 10 or Linux. You could also dual boot linux on Windows 7 if you have enough internal storage.

Your other point about Prime Video, unfortunately it sounds like they use DRM.

**Comment by GrampaWildWilly** [comment]
I don't remember now but isn't there an option in the operating system for allowing file names longer than 8.3?  I ran W7 64-bit for over a decade.  Why are you downloading the 32-bit version of the CoApp?  Are you really running W7 32-bit?  If you are running 32-bit, then that indeed was the right version of the CoApp to install.

The length of the file names has me baffled.  Like I say, I ran W7 64-bit for a long time.  When I looked at file names in the system file manager, I had lots of files with long names.  What's the situation with your file names?  Also, I ran the 64-bit CoApp on W7, so it does work on W7.  But you do have to be running 64-bit Windows or it won't work.

Matt's comment about having to upgrade eventually is exactly what I faced late last year.  I finally abandoned W7 in favor of W11 only just this past December.  The final straw for me was Firefox saying they were dropping W7 support.  Otherwise, W7 ran just fine.  But you have to face reality that you're not going to be able to stick with W7 forever.  In addition, I wanted to try using an internal SSD & adding some RAM.  But my hardware was so old that I couldn't find compatible parts to make those improvements.  So it was a combination of obsolete hardware running an unsupported operating system that got me up to the modern era.  You will also have to face that decision sooner or later.

**Comment by andherson17** [comment]
Hey folks. Thanks for your input. This old W7 64-bit PC has been running like a top for around 12 years at this point. I did have a hard drive go south around 2016 when W10 came out that led to installing a fresh W7 installation. And as I wrote, I DO have a notebook running W10 that I have plugged into an HDTV I use as a monitor, and switch between the PC & notebook. So I've been using W10 since 2016, but I've never warmed up to it. I find filesystem operations far more friendly in W7, and several other aspects of W7 much more intuitive. I've actually had no major problems since MS stopped supporting it. This issue installing the occasional piece of software that complains about not being able to write to appdata\local\temp, and having to shorten the installed filename has just popped up recently. Otherwise I've had no other issues with long filenames. I'll post over at the SevenForums if it gets really bad.

@GrampaWildWilly: The PC is running 64 bit W7, but after I got the 64 bit copy of  CoApp installed, the Chrome VD addon prompted me to upgrade CoAPP and wouldn't allow me to download my video. That's why I uninstalled CoApp 64 bit and tried the 32 bit version. VD then was happy and successfully downloaded the video.

I think @Mattmjs is right about Prime Video using DRM as an explanation about why VD's downloaded MP4s end up corrupted. As I said, that was the case with the addon on my W10 notebook. I see mentions of Pazu and Kigo subscription downloaders I might have a look at.

But yeah... I expect this old W7 PC will eventually develop too many issues to be able to deal with at some point, But till then the old beast seems to be chugging along amazingly well.

**Reply by GrampaWildWilly** [reply]
So you are on 64-bit W7.  Well, that makes the long file name issue doubly weird.  Weirder still that 64-bit CoApp doesn't work but 32-bit does.  I suppose you can't argue with success.  As for W10, I skipped it.  Went straight from W7 to W11.  And it wasn't an upgrade.  I built a new machine & did a W11 install into that bare machine.  The W7 & W11 forum sites are indispensable.  They both gave me a tremendous lot of help, especially that guy Brink.  If there's an answer out there, those are the places to be looking & asking.
